tests/unit/legate/util/test_system.py

```diff
@@ -133,49 +133,54 @@ def test_errors(self) -> None:
-
-    @pytest.mark.parametrize("val", ("2,1", "8,5,3,2", "1,3,2,5,4,7,6"))
-    def test_individual_ordered(self, val: str) -> None:
-        expected = {
-            "2,1": (1, 2),
-            "8,5,3,2": (2, 3, 5, 8),
-            "1,3,2,5,4,7,6": (1, 2, 3, 4, 5, 6, 7),
-        }
-        assert m.extract_values(val) == expected[val]
-
-    @pytest.mark.parametrize("val", ("0-2", "0-2,4-5", "0-1,3-5,8-11"))
-    def test_range(self, val: str) -> None:
-        expected = {
-            "0-2": (0, 1, 2),
-            "0-2,4-5": (0, 1, 2, 4, 5),
-            "0-1,3-5,8-11": (0, 1, 3, 4, 5, 8, 9, 10, 11),
-        }
-        assert m.extract_values(val) == expected[val]
-
-    @pytest.mark.parametrize("val", ("2-3,0-1", "0-1,4-5,2-3"))
-    def test_range_ordered(self, val: str) -> None:
-        expected = {
-            "2-3,0-1": (0, 1, 2, 3),
-            "0-1,4-5,2-3": (0, 1, 2, 3, 4, 5),
-        }
-        assert m.extract_values(val) == expected[val]
-
-    @pytest.mark.parametrize(
-        "val", ("0,1-2", "1-2,0", "0,1-2,3,4-5,6", "5-6,4,1-3,0")
-    )
-    def test_mixed(self, val: str) -> None:
-        expected = {
-            "0,1-2": (0, 1, 2),
-            "1-2,0": (0, 1, 2),
-            "0,1-2,3,4-5,6": (0, 1, 2, 3, 4, 5, 6),
-            "5-6,4,1-3,0": (0, 1, 2, 3, 4, 5, 6),
-        }
-        assert m.extract_values(val) == expected[val]
+    testdata_individual = [
+        ("0", (0,)),
+        ("1,2", (1, 2)),
+        ("3,5,7", (3, 5, 7)),
+    ]
+
+    @pytest.mark.parametrize("val,expected", testdata_individual)
+    def test_individual(self, val: str, expected: tuple[int, ...]) -> None:
+        assert m.extract_values(val) == expected
+
+    testdata_individual_ordered = [
+        ("2,1", (1, 2)),
+        ("8,5,3,2", (2, 3, 5, 8)),
+        ("1,3,2,5,4,7,6", (1, 2, 3, 4, 5, 6, 7)),
+    ]
+
+    @pytest.mark.parametrize("val,expected", testdata_individual_ordered)
+    def test_individual_ordered(
+        self, val: str, expected: tuple[int, ...]
+    ) -> None:
+        assert m.extract_values(val) == expected
+
+    testdata_range = [
+        ("0-2", (0, 1, 2)),
+        ("0-2,4-5", (0, 1, 2, 4, 5)),
+        ("0-1,3-5,8-11", (0, 1, 3, 4, 5, 8, 9, 10, 11)),
+    ]
+
+    @pytest.mark.parametrize("val,expected", testdata_range)
+    def test_range(self, val: str, expected: tuple[int, ...]) -> None:
+        assert m.extract_values(val) == expected
+
+    testdata_range_ordered = {
+        ("2-3,0-1", (0, 1, 2, 3)),
+        ("0-1,4-5,2-3", (0, 1, 2, 3, 4, 5)),
+    }
+
+    @pytest.mark.parametrize("val,expected", testdata_range_ordered)
+    def test_range_ordered(self, val: str, expected: tuple[int, ...]) -> None:
+        assert m.extract_values(val) == expected
+
+    testdata_mixed = {
+        ("0,1-2", (0, 1, 2)),
+        ("1-2,0", (0, 1, 2)),
+        ("0,1-2,3,4-5,6", (0, 1, 2, 3, 4, 5, 6)),
+        ("5-6,4,1-3,0", (0, 1, 2, 3, 4, 5, 6)),
+    }
```

Comment:
Only comment: some are lists of tuples and some are sets of tuples? FWIW I think you could make the `testdata_foo` values all be dicts, and then pass e.g. `testdata_foo.items()` to the `parametrize` call

Reply:
Good catch. Fixed.